src/hooks/useEvm/useEvm.ts

```diff
@@ -1,61 +1,70 @@
 import type { ETHWallet } from '@shapeshiftoss/hdwallet-core'
 import { useCallback, useEffect, useMemo, useState } from 'react'
 import { useWallet } from 'hooks/useWallet/useWallet'
-import { bn } from 'lib/bignumber/bignumber'
 import { getSupportedEvmChainIds } from 'lib/utils/evm'
 import { selectFeatureFlags } from 'state/slices/preferencesSlice/selectors'
 import { useAppSelector } from 'state/store'
 
-const chainIdFromEthNetwork = (
-  ethNetwork: string | undefined,
+const evmChainIdFromChainId = (
+  maybeEvmChainId: ChainId | undefined,
   supportedEvmChainIds: ChainId[],
 ): ChainId | undefined => {
-  if (!ethNetwork) return
-
-  return supportedEvmChainIds.find(chainId => fromChainId(chainId).chainReference === ethNetwork)
+  if (!maybeEvmChainId) return
+  return supportedEvmChainIds.find(chainId => chainId === maybeEvmChainId)
 }
 
 export const useEvm = () => {
   const {
     state: { wallet },
   } = useWallet()
   const [isLoading, setIsLoading] = useState<boolean>(true)
-  const [ethNetwork, setEthNetwork] = useState<string>()
+  const [chainId, setChainId] = useState<ChainId>()
   const featureFlags = useAppSelector(selectFeatureFlags)
   const supportedEvmChainIds = useMemo(
     () => getSupportedEvmChainIds(),
     // We want to explicitly react on featureFlags to get a new reference here
     // eslint-disable-next-line react-hooks/exhaustive-deps
     [featureFlags],
   )
-  const getChainIdFromEthNetwork = useCallback(
-    (ethNetwork: string) => chainIdFromEthNetwork(ethNetwork, supportedEvmChainIds),
+  const isEvmChainId = useCallback(
+    (chainId: ChainId) => evmChainIdFromChainId(chainId, supportedEvmChainIds) !== undefined,
     [supportedEvmChainIds],
   )
 
   useEffect(() => {
     ;(async () => {
       setIsLoading(true)
-      const ethNetwork = await (wallet as ETHWallet)?.ethGetChainId?.()
-      ethNetwork && setEthNetwork(bn(ethNetwork).toString())
+      const maybeChainReference = (await (wallet as ETHWallet)?.ethGetChainId?.())?.toString() ?? ''
+      const chainNamespace = CHAIN_NAMESPACE.Evm
+
+      if (isChainReference(maybeChainReference)) {
+        setChainId(
+          toChainId({
+            chainNamespace,
+            chainReference: maybeChainReference,
+          }),
+        )
+      }
     })()
   }, [wallet])
 
   const connectedEvmChainId = useMemo(() => {
-    if (ethNetwork && isLoading) {
+    if (chainId && isLoading) {
       setIsLoading(false)
     }
 
-    return chainIdFromEthNetwork(ethNetwork, supportedEvmChainIds)
-  }, [isLoading, ethNetwork, supportedEvmChainIds])
+    return evmChainIdFromChainId(chainId, supportedEvmChainIds)
+  }, [isLoading, chainId, supportedEvmChainIds])
 
-  return {
-    connectedEvmChainId,
-    getChainIdFromEthNetwork,
-    isLoading,
-    setEthNetwork,
-    supportedEvmChainIds,
-  }
+  return useMemo(
+    () => ({
+      connectedEvmChainId,
+      isEvmChainId,
+      isLoading,
+      setEthNetwork: setChainId,
```

Comment:
Rugged by LSP replace
```suggestion
      setChainId,
```

Reply:
a938f56d3beb4fbbaa8ad56aa222bbe78eece622